File: component-starter-server/src/main/frontend/src/Component/Input.js
/**
 *  Copyright (C) 2006-2017 Talend Inc. - www.talend.com
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
import React from 'react';

import theme from './Input.scss';

export default class Input extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: this.props.initialValue
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(evt) {
    let diff = {};
    if (!!this.props.aggregate) {
      this.props.aggregate[this.props.accessor] = evt.target.value;
    } else {
      diff.value = evt.target.value;
    }
    diff.message = evt.target.checkValidity() ? undefined: evt.target.validationMessage;
    if (!diff.message) {
      this.props.onChange && this.props.onChange(evt.target.value);
    }
    this.setState(diff);
  }

  render() {
    return (
      <span className={theme.Input}>
        <input {...this.props} onChange={e => this.onChange(e)} value={!!this.props.aggregate ?  this.props.aggregate[this.props.accessor] : this.state.value}/>
        {
          !!this.state.message && <span className={theme.error}>{this.state.message}</span>
        }
      </span>
    );
  }
}

File: component-runtime-testing/component-runtime-junit/src/test/java/org/talend/sdk/component/test/Input.java
/**
 * Copyright (C) 2006-2018 Talend Inc. - www.talend.com
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.talend.sdk.component.test;

import java.io.Serializable;

import org.talend.sdk.component.api.input.Emitter;
import org.talend.sdk.component.api.input.Producer;
import org.talend.sdk.component.api.record.Record;

@Emitter(name = "Input")
public class Input implements Serializable {

    @Producer
    public Record next() {
        return null;
    }
}
